src/shared/components/test/Buttons-test.js

```diff
@@ -0,0 +1,343 @@
+import { mount } from 'enzyme';
+
+import { IconButton, LabeledButton, LinkButton } from '../Buttons.js';
+import { $imports } from '../Buttons.js';
+
+//import { checkAccessibility } from '../../../test-util/accessibility';
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+
+describe('IconButton', () => {
+  let fakeOnClick;
+
+  function createComponent(props = {}) {
+    return mount(
+      <IconButton
+        icon="fakeIcon"
+        title="My Action"
+        onClick={fakeOnClick}
+        {...props}
+      />
+    );
+  }
+
+  beforeEach(() => {
+    fakeOnClick = sinon.stub();
+    $imports.$mock(mockImportedComponents());
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('renders the indicated icon', () => {
+    const wrapper = createComponent();
+
+    const icon = wrapper.find('SvgIcon');
+    assert.equal(icon.prop('name'), 'fakeIcon');
+  });
+
+  it('invokes callback on click', () => {
+    const onClick = sinon.stub();
+    const wrapper = createComponent({ onClick });
+
+    wrapper.find('button').simulate('click');
+    assert.calledOnce(onClick);
+  });
+
+  it('uses a default className', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(wrapper.find('button').hasClass('IconButton'));
+  });
+
+  it('renders a primary variant', () => {
+    const wrapper = createComponent({ variant: 'primary' });
+
+    assert.isTrue(wrapper.find('button').hasClass('IconButton--primary'));
+  });
+
+  it('overrides className when provided', () => {
+    const wrapper = createComponent({
+      className: 'CustomClassName',
+      variant: 'primary',
+    });
+
+    assert.isTrue(wrapper.find('button').hasClass('CustomClassName'));
+    assert.isTrue(wrapper.find('button').hasClass('CustomClassName--primary'));
+    assert.isFalse(wrapper.find('button').hasClass('IconButton'));
+  });
+
+  [
+    {
+      propName: 'expanded',
```

Comment:
a slightly less wordy way of doing this (as a suggestion)


```
{
     prop: ['expanded', true]
     attribute:  ['aria-expanded', true]
},
```

Reply:
Now that this isn't duplicated throughout the test file, I don't think it's quite as problematic.